neurodamus/connection_manager.py

```diff
@@ -728,16 +730,32 @@ def _get_conn_stats(self, _src_target, dst_target):
           - _src target is not considered so we count all inbound synapses
           -  We will only consider gids which have not been accounted for yet.
         """
+        BLOCK_LENGTH = 5000
+        SAMPLE_SIZE = 100
         raw_gids = dst_target.get_local_gids(raw_gids=True) if dst_target else self._raw_gids
         new_gids = numpy.setdiff1d(raw_gids, self._dry_run_counted_cells, assume_unique=True)
-        if len(new_gids):
-            counts = self._synapse_reader.get_counts(new_gids, group_by="syn_type_id")
-            self._dry_run_counted_cells = numpy.union1d(self._dry_run_counted_cells, new_gids)
-            src_pop = self._src_cell_manager.population_name
-            target = (dst_target.name if dst_target else f"*{self._cell_manager.population_name}")
-            logging.debug("(rank0) %s -> %s %s: %s", src_pop, target, new_gids, counts)
-            return counts
-        return {}
+        if not len(new_gids):
+            return {}
+
+        temp_counter = Counter()
+        chunking_gen = gen_ranges(len(new_gids), BLOCK_LENGTH)
+        n_blocks = ceil(len(new_gids)/BLOCK_LENGTH)
+        src_pop = self._src_cell_manager.population_name
+        target = (dst_target.name if dst_target else f"*{self._cell_manager.population_name}")
+        for cycle, (low, high) in enumerate(chunking_gen):
+            sample_gids = new_gids[low:low + SAMPLE_SIZE]
+            sample_counts = self._synapse_reader.get_counts(sample_gids, group_by="syn_type_id")
+
+            extrapolated_counts = {
+                syn_type: count * (min(high, len(new_gids)) - low) / SAMPLE_SIZE
+                for syn_type, count in sample_counts.items()}
+            temp_counter.update(extrapolated_counts)
+            if cycle > 10:
+                log_all(VERBOSE_LOGLEVEL, "Rank: %d, %.2f%%", MPI.rank, (cycle+1)/n_blocks*100)
+        self._dry_run_counted_cells = numpy.union1d(self._dry_run_counted_cells, new_gids)
```

Comment:
`_dry_run_counted_cells` contains the cells already considered. As it is we consider the whole range as counted, which I think is correct, but have an extra though about it.

Reply:
I've extracted the extrapolation from the loop. Hope it makes the code simpler and more efficient.

Reply:
I think it makes sense to flag them all as counted. After all we've already taken a sample from that block so, in the logic of not being able to count them all, they should be flagged as counted anyway.

Is there any particular reason why we shouldn't do so?

Reply:
The only reason I could think about is, when handling "connection blocks" which overlap we would ensure some of their cells would be taken into account. But... it gets tricky to get the extrapolated counts right. We probably better trying to correctly sample over the whole range. It makes me think that we probably want to sample at regular intervals 🤔